### DISCUSSION TITLE
Is there a way to use T2I adapters with an image2image pipeline?

### DISCUSSION THREAD

**Discussion by nivedwho** [huggingface/diffusers#7890] [Q&A]
I want to use T2I canny adapter along with additional image inputs passed along with strength value like image2image pipeline? Is there a SDXLAdapterImg2ImgPipline or something similar that achieves the same thing?

**Comment by tolgacangoz** [comment]
Aren't T2I-Adapters special for **T**ext-**2**-**I**mage :thinking:? [There is a multi-adapter usage that you can adjust conditioning levels.](https://huggingface.co/docs/diffusers/main/en/using-diffusers/t2i_adapter#multiadapter)
[Or, does Image2Image with IP-Adapters help?](https://huggingface.co/docs/diffusers/main/en/using-diffusers/ip_adapter?tasks=Image-to-image#general-tasks)

**Reply by asomoza** [reply]
Not really, they're named like that but it's just a form of control. There isn't a reason why they can't be used with image-to-image, it's just that no one has created the community pipeline.

|source|normal|canny t2i adapter|
|---|---|---|
|![20240501130625_2487854446](https://github.com/huggingface/diffusers/assets/5442875/ce048caf-9a7e-4924-aa45-a4fc769fda96)|![ComfyUI_00100_](https://github.com/huggingface/diffusers/assets/5442875/359f8f7f-6cbe-4766-af27-a319e7ed3244)|![ComfyUI_00101_](https://github.com/huggingface/diffusers/assets/5442875/789c7b0b-b27a-4faa-8f3c-7309dc1c4d10)|

Since @nivedwho is asking for something similar, the more obvious one is the controlnet equivalent: [StableDiffusionXLControlNetImg2ImgPipeline ](https://github.com/huggingface/diffusers/blob/main/src/diffusers/pipelines/controlnet/pipeline_controlnet_sd_xl_img2img.py)

**Reply by nivedwho** [reply]
Yes, I actually wanted to use the T2I adapter instead of the existing controlnet img2img pipe just to save some inference time.

**Reply by tolgacangoz** [reply]
@asomoza Hmm, thank you for your answer. I will look into this.

@nivedwho [Faster diffusion on less beefy GPUs](https://github.com/huggingface/diffusers/discussions/6609). You can also [compile ControlNet](https://huggingface.co/docs/diffusers/main/en/optimization/torch2.0#controlnet).
